app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,26 @@
+class CategoriesController < ApplicationController
+  def index
+    @categories = Category.where(author: current_user)
+    @expense = Expense.where(author: current_user)
+    @expense_categories = ExpenseCategory.all
+  end
+
+  def new
+    @new_category = Category.new
+  end
+
+  def create
+    @create_category = Category.create(create_category_params)
+    if @create_category.save
+      redirect_to categories_path
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def create_category_params
+    params.permit(:name, :icon, :author_id)
+  end
+end
```

Comment:
- [ ] [REQUIRED] Currently I am unable to add a new category to the application. The application throws an Unpermitted parameters error because there is an extra param in the `create_category_params` but we are only passing the name and icon.

### Here is the error
<img width="891" alt="Screenshot 2023-04-28 at 19 06 04" src="https://user-images.githubusercontent.com/58692590/235223924-b7dddf0d-4595-4dd0-ad24-582ea49fd82c.png">


### To fix the error
Please modify the `create_category_params` method to permit only the desired parameters:
``` ruby
def create_category_params
  params.permit(:name, :icon)
end
```
By removing `:author_id` from the permitted parameters list, the error should no longer occur.

Reply:
I removed :author_id and my code breaks. Kindly use the name of the images in the image folder.
![images](https://user-images.githubusercontent.com/109760801/235283291-b324e402-f85f-460e-a497-a4dda540296d.JPG)

Reply:
- Great implementation @Harrynoqs 👍🏽